File: src/building blocks/BTG.Core/DomainObjects/PagedResult.cs
using System.Collections.Generic;

namespace BTG.Core.DomainObjects
{
    public class PagedResult<T> where T : class
    {
        public IEnumerable<T> List { get; set; }
        public int TotalResults { get; set; }
        public int PageIndex { get; set; }
        public int PageSize { get; set; }
        public string Nome { get; set; }
        public double TotalPages => Math.Ceiling((double)TotalResults / PageSize);
        public PagedResult()
        {
            TotalResults = 0;
            PageIndex = 1;
            PageSize = 0;
        }
    }
}

File: src/web/btg-app/src/app/models/paged-result.ts
export class PagedResult<T> {
    list: T[]
    totalResults: number;
    pageIndex: number;
    pageSize: number;
    totalPages: number;
  
    constructor(){
      this.list = [];
      this.totalResults = 0;
      this.pageIndex = 1;
      this.pageSize = 6;
      this.totalPages = 1
    }
  }